File: frontend/src/app/models/fakultet.ts
import { Adresa } from "./adresa";
import { Nastavnik } from "./nastavnik";
import { Univerzitet } from "./univerzitet";
export interface FakultetDTO{
  id:number|null,
  naziv:string,
  opis:string,
  email:string
  kontakt:string,
  adresa:Adresa,
  rektor:Nastavnik,
  univerzitetId:number,
}

File: admin-service/src/main/java/com/projekat/admin_service/entity/Fakultet.java
package com.projekat.admin_service.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Fakultet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String naziv;
    private String opis;
    private String kontakt;
    private String email;
    private Boolean aktivan;
    @ManyToOne
    private Univerzitet univerzitet;
    private long nastavnikId;
    @OneToOne(cascade = CascadeType.ALL)
    private Adresa adresa;
}

